### DISCUSSION TITLE
play function getByRole loses DOM reference

### DISCUSSION THREAD

**Discussion by aaron-hartnell** [storybookjs/storybook#25197] [Help]
### Summary

Hi I have some tests that refuse to pass and I really don't understand why. I launch storybook in devmode and when I go to the story the first time the interaction test fails. If ever I modify and recompile the code the hot-loading passes the test without any problem. 
The bug consists on a userEvent.click on a button that doesn't fire. (It fires when the hot-reload loads the story or when I replay the test in the interaction panel). If I console.log the button I get it's reference in the console without any problem but the reference (even thow it's the right button) isn't refering to the one on screen. On hot reload or replay interaction, it does.

If I add and await a sleep function for like 100 ms in the beginning of the function eveything works fine. 

What I really don't get is why does the function catch the reference ut not the one shown on screen.

**Comment by Marklb** [comment]
It is difficult to say, without knowing anything about the code.

Can you create a reproduction on Stackblitz (https://new-storybook.netlify.app/) or a repo that can be cloned?

What framework are you using?

My best guess: Does your component create elements outside the component, such as an overlay attached to the body, that may not get cleaned up and still may be in the DOM after a re-render? Elements created after a delay that don't get cleaned up on component destroy is my only guess, without seeing any code.
